### DISCUSSION TITLE
Question: Does ServiceStack support content type-"multipart/form-data"?

### DISCUSSION THREAD

**Discussion by sahanar28** [ServiceStack/Discuss#68] [Q&A]
Does ServiceStack support content type-"multipart/form-data"? Is ServiceStack capable to register "multipart/form-data" as a content type?

**Comment by mythz** [comment]
Yes you can upload files to any API where it will be available in `Request.Files` collection. 
We also have a [Managed File Uploads](https://docs.servicestack.net/locode/files-overview) solution that further simplifies file uploads.

**Reply by sahanar28** [reply]
But In the ServiceStack documentation it doesn't explicitly mention the ability to register 'multipart/form-data'. This suggests that direct support for handling "multipart/form-data" might not be a built-in feature of ServiceStack. Is this correct?

**Reply by mythz** [reply]
It's not an add-on, it's already built into all HTTP Servers ServiceStack runs on, `Request.Files` simply exposes the uploaded file collection that's made available by the underlying HTTP Server e.g. ASP .NET Core / classic ASP.NET for .NET Framework.

**Reply by sahanar28** [reply]
We are having an issue uploading a large file using an endpoint.

Our endpoint uses ServiceStack library.

ServiceStack by default will set content type to "application/json" when uploading files? Correct?

We want to know if it's possible to shift the endpoint into using "multipart/form-data"? 

If it's possible to register multipart/form data and change content type from "application/json" to "multipart/form-data" and force it to be used by our API?

what are the steps to follow?
Is there any recommended approach?

**Reply by mythz** [reply]
Please refer to the docs for [File Upload examples](https://docs.servicestack.net/locode/files-overview#uploading-files-from-c). 

Inside your Service uploaded files are available from the `IRequest.Files` collection, which will be up to your service implementation to handle.
